gst/nnstreamer/tensor_filter/tensor_filter_custom.c

```diff
@@ -68,24 +68,27 @@ custom_loadlib (const GstTensorFilterProperties * prop, void **private_data)
 
   if (*private_data != NULL) {
     /** @todo : Check the integrity of filter->data and filter->model_file, nnfw */
-    return 1;
+    ml_loge ("Init is called but it is already initialized.\n");
+    return -EINVAL;
   }
 
   if (!prop->model_files || prop->num_models != 1 ||
       !prop->model_files[0] || prop->model_files[0][0] == '\0') {
     /* The .so file path is not given */
-    return -1;
+    ml_logw ("Custom filter file is not given.\n");
+    return -EINVAL;
   }
 
   if (!nnsconf_validate_file (NNSCONF_PATH_CUSTOM_FILTERS,
           prop->model_files[0])) {
     /* Cannot load the library */
-    return -1;
+    ml_logw ("Custom filter file %s is invalid.\n", prop->model_files[0]);
+    return -EINVAL;
   }
 
   ptr = *private_data = g_new0 (internal_data, 1);
   if (ptr == NULL) {
-    ml_loge ("Failed to allocate memory for custom filter.");
+    ml_loge ("Failed to allocate memory for custom filter.\n");
     return -1;
```

Comment:
It seems to be better to change this return value as well.

Reply:
Updated (ENOMEM)